### PR TITLE
User/jmart/add events doc

### PR DESCRIPTION
**Pull request by James-Mart**
This PR breaks the wallet applet, front-end needs to be updated to use the graphql interface, rather than rest

### CODE REVIEW

**Comment by James-Mart** [comment]
* `getTokenTypes` rest query must be replaced by the graphql query:
```
query {
    tokenTypes
    {
        pageInfo {
            hasNextPage
            endCursor
        }
        edges {
            node {
                id
                precision {
                    value
                }
                symbolId
            }
        }
    }
}
```
* `getUserConf` rest query must be replaced by the graphql query: 
```
query {
    getUserConf(user: "alice", flag: "manualDebit" )
}
```
* `balances` rest query must be replaced by the graphql query:
```
query {
    userBalances(user: "alice") {
        key {
            account
            tokenId
        }
        balance
    }
}
```

File: doc/psidk/src/concepts/events.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+| 7   | `4`       | ...   | ...   | ...  |
+
+- event head = 7
+- event chain = 7, 4, 2
+
+<br/>
+
+For example, in the example Token Service, a `Transferred` event is defined, and is included in an index of events called `UserEvents`:
+
+<details>
```
Comment (tbfleming) [inline doc/psidk/src/concepts/events.md:65]: Nit pick: I'm not a fan of pages that require you to click to reveal

File: libraries/psibase/common/include/psibase/eventIndex.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace psibase
+{
+   // This does nothing, but causes an error when called from a `consteval` function.
+   inline void expectedNullTerminatedArray() {}
+
+   // StringLiteral idea taken from:
+   // https://stackoverflow.com/questions/68024563/passing-a-string-literal-to-a-template-char-array-parameter
+   template <size_t N>
+   struct StringLiteral
```
Comment (tbfleming) [inline libraries/psibase/common/include/psibase/eventIndex.hpp:11]: Redundant with `psio::FixedString`

File: services/system/AuthEcSys/src/RAuthEcSys.cpp [hunk 3]
```diff
@@ -9,16 +10,45 @@
...
+   struct AuthEcQuery
+   {
+      auto accWithKey(psibase::PublicKey key) const
+      {
+         auto idx = AuthEcSys::Tables{AuthEcSys::service}
+                        .open<AuthEcSys::AuthTable>()
+                        .getIndex<1>()
+                        .subindex(key);
+
+         optional<std::tuple<AccountNumber>> gt;
```
Comment (tbfleming) [inline services/system/AuthEcSys/src/RAuthEcSys.cpp:29]: Make these args of `accWithKey`

Comment (tbfleming) [reply services/system/AuthEcSys/src/RAuthEcSys.cpp:29]: But without tuple or the user side of the query gets icky. Unfortunately you'll have to translate between `optional<AccountNumber>` and `optional<tuple<AccountNumber>>`

**Review by tbfleming — APPROVED** [review]

**Pull request merged**